### PR TITLE
example/centos: import OCI

### PR DESCRIPTION
**Pull request by supakeen**
Generate reference manifests for the OCI image and do an initial implementation of the omnifests. Based on #219.

### CODE REVIEW

**Comment by supakeen** [comment]
Rebased now that #219 has landed.

File: example/centos/centos-9-x86_64-oci.yaml [hunk 1]
```diff
@@ -0,0 +1,174 @@
+otk.version: "1"
+
+otk.define:
+  filesystem:
+    modifications:
```
Comment (mvo5) [inline example/centos/centos-9-x86_64-oci.yaml:5]: (nitpick) for consistency with the other a `# empty` comment here would be good I think to indicate that we left this empty on purpose.

File: example/centos/centos-9-x86_64-oci.yaml [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+    build:
+      otk.external.osbuild-gen-depsolve-dnf4:
+        architecture: x86_64
+        module_platform_id: c9s
+        releasever: "9"
+        repositories:
+          otk.include: "common/repositories/x86_64.yaml"
+        packages:
+          include:
+            - dosfstools
```
Comment (mvo5) [inline example/centos/centos-9-x86_64-oci.yaml:16]: (nitpick) this is not sorted

**Review by mvo5 — COMMENTED** [review]
This is a curious one, for my own understanding, why is this called "oci"? This is almost identical to the qcow2 image:
```yaml
 --- centos-9-x86_64-oci.yaml	2024-09-30 15:24:52.030512373 +0200
+++ centos-9-x86_64-qcow2.yaml	2024-09-30 10:44:54.891368871 +0200
@@ -3,6 +3,7 @@
 otk.define:
   filesystem:
     modifications:
+    # empty
   packages:
     build:
       otk.external.osbuild-gen-depsolve-dnf4:
@@ -13,8 +14,8 @@
           otk.include: "common/repositories/x86_64.yaml"
         packages:
           include:
-            - dosfstools
             - coreutils
+            - dosfstools
             - glibc
             - grub2-pc
             - platform-python
@@ -24,8 +25,8 @@
             - rpm
             - selinux-policy-targeted
             - systemd
-            - xz
             - xfsprogs
+            - xz
           exclude: []
     os:
       otk.external.osbuild-gen-depsolve-dnf4:
@@ -78,8 +79,8 @@
             - firewalld
             - iprutils
             - ivtv-firmware
-            - iwl1000-firmware
             - iwl100-firmware
+            - iwl1000-firmware
             - iwl105-firmware
             - iwl135-firmware
             - iwl2000-firmware
@@ -143,14 +144,14 @@
               no_zero_conf: true
         - otk.external.otk-make-fstab-stage:
             ${filesystem}
-        - otk.include: fragment/grub2/x86_64.yaml
+        - otk.include: "fragment/grub2/x86_64.yaml"
         - type: org.osbuild.systemd
           options:
             default_target: multi-user.target
         - type: org.osbuild.selinux
           options:
             file_contexts: etc/selinux/targeted/contexts/files/file_contexts
-    - otk.include: pipeline/image/x86_64.yaml
+    - otk.include: "pipeline/image/x86_64.yaml"
     - name: qcow2
       build: name:build
       stages:
```

File: example/centos/centos-9-x86_64-oci.yaml (hunk 1 above)
Comment (supakeen) [reply example/centos/centos-9-x86_64-oci.yaml:5]: Done.

File: example/centos/centos-9-x86_64-oci.yaml (hunk 2 above)
Comment (supakeen) [reply example/centos/centos-9-x86_64-oci.yaml]: Done.

**Comment by supakeen** [comment]
> This is a curious one, for my own understanding, why is this called "oci"? This is almost identical to the qcow2 image.

Oracle Cloud Infrastructure. It is a separate image type instead of an alias so it shows up in (some) APIs. In `otk` I *guess* it could be a symlink or single include of the qcow2 if you prefer.

**Comment by mvo5** [comment]
> > This is a curious one, for my own understanding, why is this called "oci"? This is almost identical to the qcow2 image.
> 
> Oracle Cloud Infrastructure. It is a separate image type instead of an alias so it shows up in (some) APIs. In `otk` I _guess_ it could be a symlink or single include of the qcow2 if you prefer.

Thanks! I would love this to be a symlink, yes (or an include) that seems like an easy win :)

**Comment by supakeen** [comment]
I have rewritten the single commit here to only import the reference manifest and symlink the omnifest to the qcow2.

**Review by mvo5 — APPROVED** [review]
\o/ thank you

**Pull request merged**